vars/detectExecuteScan.groovy

```diff
@@ -0,0 +1,136 @@
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = [
+    /**
+     * Defines the tool which is used for building the artifact.<br />
+     * Currently, it is possible to select two behaviors of the step:
+     *
+     * 1. Golang-specific behavior (`buildTool: golang`). Assumption here is that project uses the dependency management tool _dep_
+     * 2. Custom-specific behavior for all other values of `buildTool`
+     *
+     * @possibleValues `golang`, any other build tool
+     */
+    'buildTool',
+    'detect',
+    /**
+     * Name of the Synopsis Detect (formerly BlackDuck) project.
+     * @parentConfigKey detect
+     */
+    'projectName',
+    /**
+     * Version of the Synopsis Detect (formerly BlackDuck) project.
+     * @parentConfigKey detect
+     */
+    'projectVersion',
+    /**
+     * Properties passed to the Synopsis Detect (formerly BlackDuck) scan. You can find details in the [Synopsis Detect documentation](https://synopsys.atlassian.net/wiki/spaces/INTDOCS/pages/622846/Using+Synopsys+Detect+Properties)
+     * @parentConfigKey detect
+     */
+    'scanProperties',
+    /**
+     * Server url to the Synopsis Detect (formerly BlackDuck) Server.
+     * @parentConfigKey detect
+     */
+    'serverUrl',
+    /**
+     * Jenkins 'Secret text' credentials ID containing the API token used to authenticate with the Synopsis Detect (formerly BlackDuck) Server.
+     * @parentConfigKey detect
+     */
+    'userTokenCredentialsId'
```

Comment:
Shouldn't this be renamed to reflect the reference to an API token?

Reply:
changed